src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.IO.cs

```diff
@@ -160,17 +167,43 @@ private async Task WriteBody(bool flush = false)
                 }
             }
             // We want to swallow IO exception and allow app to finish writing
+            catch (ConnectionResetException)
+            {
+                ConnectionReset();
+            }
             catch (Exception ex)
             {
-                if (!(ex is IOException))
-                {
-                    _bodyOutput.Reader.Complete(ex);
-                }
+                error = ex;
             }
             finally
             {
-                _bodyOutput.Reader.Complete();
+                _bodyOutput.Reader.Complete(error);
             }
         }
+
+        private void CompleteIO()
+        {
+            if (Interlocked.CompareExchange(ref _requestAborted, 1, 0) != 0)
+            {
+                return;
+            }
+
+            _bodyOutput.Dispose();
+            CancelRequestAbortedToken();
+        }
+
+        public void Abort(Exception reason)
+        {
+            _bodyOutput.Abort(reason);
+            _streams.Abort(reason);
+            NativeMethods.HttpCloseConnection(_pInProcessHandler);
+
+            CompleteIO();
+        }
+
+        private void ConnectionReset()
```

Comment:
Remove ConnectionReset and just call CompleteIO.

Reply:
I'd like to keep it for future logging changes